stac_fastapi/types/stac_fastapi/types/search.py

```diff
@@ -1,79 +1,23 @@
 """stac_fastapi.types.search module.
 
-# TODO: replace with stac-pydantic
 """
 
 import abc
-import operator
-from datetime import datetime
-from enum import auto
-from types import DynamicClassAttribute
-from typing import Any, Callable, Dict, Generator, List, Optional, Union
+from typing import Dict, List, Optional
 
 import attr
-from geojson_pydantic.geometries import (
-    LineString,
-    MultiLineString,
-    MultiPoint,
-    MultiPolygon,
-    Point,
-    Polygon,
-    _GeometryBase,
-)
-from pydantic import BaseModel, ConstrainedInt, validator
-from pydantic.errors import NumberNotGtError
-from pydantic.validators import int_validator
-from stac_pydantic.shared import BBox
-from stac_pydantic.utils import AutoValueEnum
-
-from stac_fastapi.types.rfc3339 import rfc3339_str_to_datetime, str_to_interval
-
-# Be careful: https://github.com/samuelcolvin/pydantic/issues/1423#issuecomment-642797287
-NumType = Union[float, int]
-
-
-class Limit(ConstrainedInt):
-    """An positive integer that maxes out at 10,000."""
-
-    ge: int = 1
-    le: int = 10_000
-
-    @classmethod
-    def __get_validators__(cls) -> Generator[Callable[..., Any], None, None]:
-        """Yield the relevant validators."""
-        yield int_validator
-        yield cls.validate
-
-    @classmethod
-    def validate(cls, value: int) -> int:
-        """Validate the integer value."""
-        if value < cls.ge:
-            raise NumberNotGtError(limit_value=cls.ge)
-        if value > cls.le:
-            return cls.le
-        return value
-
-
-class Operator(str, AutoValueEnum):
-    """Defines the set of operators supported by the API."""
-
-    eq = auto()
-    ne = auto()
-    lt = auto()
-    lte = auto()
-    gt = auto()
-    gte = auto()
-
-    # TODO: These are defined in the spec but aren't currently implemented by the api
-    # startsWith = auto()
-    # endsWith = auto()
-    # contains = auto()
-    # in = auto()
-
-    @DynamicClassAttribute
-    def operator(self) -> Callable[[Any, Any], bool]:
-        """Return python operator."""
-        return getattr(operator, self._value_)
+from pydantic import PositiveInt
+from pydantic.functional_validators import AfterValidator
+from stac_pydantic.api import Search
+from typing_extensions import Annotated
```

Comment:
There is a lot of changes in this file, can you explain?

Reply:
I removed classes that are now implemented in stac_pydantic